client/me/billing-history/transactions-header.jsx

```diff
@@ -89,32 +98,32 @@ class TransactionsHeader extends React.Component {
 		);
 	}
 
-	setFilter( filter ) {
-		this.setState( { activePopover: '' } );
-		this.props.onNewFilter( filter );
-	}
-
 	renderDatePopover() {
-		const previousMonths = range( 6 ).map( function( n ) {
-			return this.props.moment().subtract( n, 'months' );
-		}, this );
+		const { filter, dateFilters } = this.props;
+		const selectedFilter = find( dateFilters, filterMeta =>
+			isEqual( filterMeta.value, filter.date )
+		);
+		const selectedText = selectedFilter ? selectedFilter.title : this.props.translate( 'Date' );
```

Comment:
Perhaps destructure `translate` from `props` as it's used more than once in this method? That applies for other `render*` methods in this PR as well.